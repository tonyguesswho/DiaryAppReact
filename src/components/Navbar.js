import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { signout } from '../actions/authAction';

class Navbar extends Component {
  logout = () => {
    signout();
  };
  render() {
    return (
      <nav>
        {!localStorage.token ? (
          <ul className="nav main-nav" id="authNav">
            <li>
              <Link to="/signin">Sign In</Link>
            </li>
            <li>
              <Link to="/signup">Sign Up</Link>
            </li>
          </ul>
        ) : (
          <ul className="nav main-nav" id="authNav">
            <li>
              <Link to="/profile">Profile</Link>
            </li>
            <li>
              <Link id="signout" to="/" onClick={this.logout}>Sign out</Link>
            </li>
          </ul>
        )}
      </nav>
    );
  }
}
export default Navbar;
