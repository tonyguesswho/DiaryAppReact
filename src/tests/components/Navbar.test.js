import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Navbar from '../../components/Navbar';

let wrapped;

beforeEach(() => {
  wrapped = shallow(<Navbar />);
});

describe('NavbarUI', () => {
  describe('render features', () => {
    test('component should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});
