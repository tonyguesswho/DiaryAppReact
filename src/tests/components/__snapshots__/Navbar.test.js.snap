// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarUI render features component should render as expected 1`] = `
<nav>
  <ul
    className="nav main-nav"
    id="authNav"
  >
    <li>
      <Link
        replace={false}
        to="/signin"
      >
        Sign In
      </Link>
    </li>
    <li>
      <Link
        replace={false}
        to="/signup"
      >
        Sign Up
      </Link>
    </li>
  </ul>
</nav>
`;
